File: backend-java/src/main/java/com/example/modules/orders/publicapi/OrdersPublicApi.java
// ONLY interface other modules/apps may import from the orders module.
package com.example.modules.orders.publicapi;

import com.example.modules.orders.internal.OrderService;
import java.util.Optional;

public class OrdersPublicApi {
    private final OrderService svc = new OrderService();
    public OrderService.Order createOrder(String orderId, long totalCents) { return svc.createOrder(orderId, totalCents); }
    public Optional<OrderService.Order> getOrder(String orderId) { return svc.getOrder(orderId); }
    public Optional<OrderService.Order> markPaid(String orderId, String paymentId) { return svc.markPaid(orderId, paymentId); }
    public Optional<OrderService.Order> markFailed(String orderId, String reason) { return svc.markFailed(orderId, reason); }
    public Optional<OrderService.Order> markRefunded(String orderId, long amountCents) { return svc.markRefunded(orderId, amountCents); }
}

File: backend-csharp/modules/orders/PublicApi/OrdersPublicApi.cs
// ONLY interface other modules/apps may import from the orders module.
using BackendCsharp.Modules.Orders.Internal;
namespace BackendCsharp.Modules.Orders.PublicApi;

public class OrdersPublicApi
{
    private readonly OrderService _svc = new();
    public Order CreateOrder(string orderId, long totalCents) => _svc.CreateOrder(orderId, totalCents);
    public Order? GetOrder(string orderId) => _svc.GetOrder(orderId);
    public Order? MarkPaid(string orderId, string paymentId) => _svc.MarkPaid(orderId, paymentId);
    public Order? MarkFailed(string orderId, string reason) => _svc.MarkFailed(orderId, reason);
    public Order? MarkRefunded(string orderId, long amountCents) => _svc.MarkRefunded(orderId, amountCents);
}
